vscode/src/debugger/session.ts

```diff
@@ -666,79 +656,28 @@ export class QscDebugSession extends LoggingDebugSession {
-                file.uri.toString(true),
-                undefined,
-                undefined,
-                "qsharp-adapter-data",
-              ),
-              this.convertDebuggerLineToClient(f.range.start.line),
-              this.convertDebuggerColumnToClient(f.range.start.character),
-            );
-            sf.endLine = this.convertDebuggerLineToClient(f.range.end.line);
-            sf.endColumn = this.convertDebuggerColumnToClient(
-              f.range.end.character,
-            );
-            return sf;
-          } else {
-            try {
-              // This file isn't part of the workspace, so we'll
-              // create a URI which can try to load it from the core and std lib
-              // There is a custom content provider subscribed to this scheme.
-              // Opening the text document by that uri will use the content
-              // provider to look for the source code.
-              const uri = vscode.Uri.from({
-                scheme: qsharpLibraryUriScheme,
-                path: f.path,
-              });
-              const source = new Source(
-                basename(f.path) ?? f.path,
-                uri.toString(),
-                0,
-                "internal core/std library",
-                "qsharp-adapter-data",
-              ) as DebugProtocol.Source;
-              const sf = new StackFrame(
-                id,
-                f.name,
-                source as Source,
-                this.convertDebuggerLineToClient(f.range.start.line),
-                this.convertDebuggerColumnToClient(f.range.start.character),
-              );
-              sf.endLine = this.convertDebuggerLineToClient(f.range.end.line);
-              sf.endColumn = this.convertDebuggerColumnToClient(
-                f.range.end.character,
-              );
-
-              return sf as DebugProtocol.StackFrame;
-            } catch (e: any) {
-              log.warn(e.message);
-              return new StackFrame(
-                id,
-                f.name,
-                undefined,
-                undefined,
-                undefined,
-              );
-            }
-          }
+          log.trace(`frames: location %O`, f.location);
+
+          const sf: DebugProtocol.StackFrame = new StackFrame(
+            id,
+            f.name,
+            new Source(
+              basename(f.location.source) ?? f.location.source,
+              f.location.source,
+              undefined,
+              undefined,
+              "qsharp-adapter-data",
+            ),
+            this.convertDebuggerLineToClient(f.location.span.start.line),
+            this.convertDebuggerColumnToClient(f.location.span.start.character),
+          );
+          sf.endLine = this.convertDebuggerLineToClient(
+            f.location.span.end.line,
+          );
+          sf.endColumn = this.convertDebuggerColumnToClient(
+            f.location.span.end.character,
+          );
+          return sf;
```

Comment:
I see a bunch of code was deleted related to debugging into the built-in core/std library. Does that "just work" now and any special handling is no longer necessary?

Reply:
Yes, you can't quite tell in this PR, but the changes in the `interpret.rs` mean that the special stdlib URLs are now returned directly from the native debugger, so we don't need to "guess" like we were doing up here.

This behavior now matches what the Language Service does (e.g. when returning URLs to stdlib in go-to-def).